### DISCUSSION TITLE
Error with Numpy 2.0.0

### DISCUSSION THREAD

**Discussion by gregorywychowaniec-zt** [jertel/elastalert2#1470] [Q&A] [answered]
Hello,

Numpy is used by Elasticsearch, and a new major upgrade (2.0.0) was release yesterday, which generates an error when creating the Elastalert indexes : ```AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?```.

This error was fixed in elasticsearch-py version 8.14.0.
So there are two ways to fix this issue: 
- Upgrade elasticsearch-py to version 8.14.0.
- Specify a version for Numpy in the requirements (1.26.4, the last version before 2.0.0).

**Comment by nsano-rururu** [comment]
Elasticsearch-py is fixed at version 7.10.1, which is compatible with both Elasticsearch and OpenSearch. When using Elastalert2, please consider using a virtual environment with venv or using Docker.

**Comment by gregorywychowaniec-zt** [comment]
Hello,

I probably misspoke. The problem comes from Elasticsearch-py which uses the latest version of Numpy in its dependencies and so is running version 2.0.0 with errors from removed type (np.float_). Those types was removed from Elasticsearch-py in version 8.14 (see [#2551](https://github.com/elastic/elasticsearch-py/pull/2551)). 
I've just proposed the two solutions to correct this problem (update Elasticsearch or fix Numpy version). 
Since the version of Elasticsearch-py doesn't seem to be able to be modified, it would be necessary to add a dependency for Numpy 1.26.4 in the Elastalert requirements file. I can create a MR with the fix if needed.

Note: I use the Docker version of Elastalert.

**Reply by nsano-rururu** [reply]
This means that we need to support a fixed version of numpy. thankyou for the explanation. Please make a pull request

**Reply by jertel** [reply]
I'm having trouble reproducing the error. When I startup ElastAlert 2 using the official latest image tag, and specifying an elastalert index of `elastalert_numpy` to force it to recreate the indices, I see no error.

```
Error response from daemon: No such container: ea
Reading Elastic 8 index mappings:
Reading index mapping 'es_mappings/8/silence.json'
Reading index mapping 'es_mappings/8/elastalert_status.json'
Reading index mapping 'es_mappings/8/elastalert.json'
Reading index mapping 'es_mappings/8/past_elastalert.json'
Reading index mapping 'es_mappings/8/elastalert_error.json'
New index elastalert_numpy created
Done!
```

**Reply by gregorywychowaniec-zt** [reply]
Can you check what version of Numpy is used during the installation of Elastalert ? Maybe it's specific to my Docker implementation

**Reply by jertel** [reply]
When using the official ElastAlert 2 Docker image there is no installation.

**Reply by gregorywychowaniec-zt** [reply]
Ok I probably miss something sorry. I would check back tomorrow.
Just for information, we use Helm integration.

My guess: "Of course there is no installation with Docker my bad, but for the next release, the problem can occur at this time when the version of Numpy in the image will change" ?

**Reply by gregorywychowaniec-zt** [reply, marked as answer]
Hello,

I can't find a reference to the installation of Numpy during the Docker build of the image and I don't have any more time to investigate. It may just be a problem with our Helm implementation. So I'm going to close the PR and if the problem occurs after the next Elastalert release, we'll know where it's coming from.
